apollo-router/src/plugins/telemetry/formatters/text.rs

```diff
@@ -153,6 +154,50 @@ impl TextFormatter {
         writer.write_char(' ')
     }
 
+    #[inline]
+    fn format_dyn_attributes<S, N>(
+        &self,
+        ctx: &FmtContext<'_, S, N>,
+        writer: &mut Writer<'_>,
+        event: &Event<'_>,
+    ) -> fmt::Result
+    where
+        S: Subscriber + for<'a> LookupSpan<'a>,
+        N: for<'a> FormatFields<'a> + 'static,
+    {
+        let span = event
+            .parent()
+            .and_then(|id| ctx.span(id))
+            .or_else(|| ctx.lookup_current());
+        if let Some(span) = span {
+            let ext = span.extensions();
+            match &ext.get::<LogAttributes>() {
+                Some(dyn_attributes) => {
+                    // TODO: put it in a method
+                    let attributes = dyn_attributes.get_attributes();
+                    // if writer.has_ansi_escapes() {
+                    //     let style = Style::new().dimmed();
+                    //     write!(writer, "{}", style.prefix())?;
+                    //     write!(writer, "[trace_id={trace_id}]")?;
+                    //     write!(writer, "{}", style.suffix())?;
+                    // } else {
+                    let attrs: Vec<String> = attributes
+                        .iter()
+                        .map(|(key, val)| format!("{key}={val}"))
+                        .collect();
+                    write!(writer, "[attributes=[{}]]", attrs.join(", "))?;
```

Comment:
this would avoid allocating a temporary string:
```suggestion
                    write!(writer, "[attributes=[")?;
                    let mut it = attributes.iter();
                    if let Some((key, value)) = it.next() {
                      write!(writer, "{key}={value}")?;
                    }
                    for (key, value) in it {
                      write!(writer, ", {key}={value}")?;
                    }
                    write!(writer, "]]")?;
```

Reply:
I know, this part is a test and I think I'll delete that part and only support it in json formatter. But if not I'll take your suggestion. Thanks